x-pack/plugin/kql/src/main/antlr/KqlBase.g4

```diff
@@ -26,70 +26,66 @@ topLevelQuery
     ;
 
 query
-    : query (AND | OR) query          #booleanQuery
-    | NOT subQuery=simpleQuery        #notQuery
-    | simpleQuery                     #defaultQuery
+    : <assoc=right> query operator=(AND | OR) query?    #booleanQuery
+    | NOT subQuery=simpleQuery                          #notQuery
+    | simpleQuery                                       #defaultQuery
     ;
 
 simpleQuery
     : nestedQuery
-    | expression
     | parenthesizedQuery
-    ;
-
-expression
-    : fieldTermQuery
-    | fieldRangeQuery
+    | matchAllQuery
+    | existsQuery
+    | rangeQuery
+    | fieldQuery
+    | fieldLessQuery
```

Comment:
`fieldTermQuery` has been split into several queries (`matchAllQuery`, `existQuery`, `fieldQuery`, `fieldLessQuery`) so a lot of conditional logic can be removed from the code.